pkg/source/ttnv3/source.go

```diff
@@ -296,3 +309,34 @@ func (s Source) setEndDevice(device *ttnpb.EndDevice, isPaths, nsPaths, asPaths,
 	}
 	return &res, nil
 }
+
+func (s Source) deleteEndDevice(ids *ttnpb.EndDeviceIdentifiers) error {
+	if address := s.config.identityServerGRPCAddress; address != "" {
```

Comment:
The order here should be in reverse (AS, NS, JS, IS) , and you are not deleting the entity from the JS.

Reply:
Reordered and added missing JS